feat: Handle paused state + workflow performance improvements

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix/issues/2569

## Changes

- Handles `paused` workflow state.
- Adds "Copy Crawl ID" and "View Archived Item" buttons to workflow detail
- Fixes file size not updating in workflow crawls list
- Fixes superadmin banner showing over workflow tabs
- Refactors workflow detail API calls to use `Task` to improve poll performance.
- Fixes execution time rendering when less than a minute

## Manual testing

1. Log in as crawler
2. Go to "Crawling"
3. Run a workflow. Verify "Latest Crawl" updates as expected
4. Click "Pause". Verify pause notices are shown.
5. Click "Dismiss". Verify pause notice is hidden.
6. Click "Resume". Verify pause notices are hidden.
7. Go to "Crawling"
8. Hover over running crawl status. Verify tooltip renders as expected.

## Screenshots

| Page | Image/video |
| ---- | ----------- |
| Workflow | <img width="538" alt="Screenshot 2025-05-14 at 3 51 19 PM" src="https://github.com/user-attachments/assets/c0e286bc-89f2-42bb-a7df-dcb837f32d6e" /> |
| Workflow (less than 10m to resume) | <img width="169" alt="Screenshot 2025-05-14 at 3 06 38 PM" src="https://github.com/user-attachments/assets/91192b22-edf0-4d57-b6a7-a9b4e0fd3f52" /> |
| Workflow | <img width="565" alt="Screenshot 2025-05-26 at 3 14 21 PM" src="https://github.com/user-attachments/assets/f535882d-1c63-4492-b338-93370aafe26d" /> |
| Workflow - Latest Crawl | <img width="970" alt="Screenshot 2025-05-14 at 4 37 33 PM" src="https://github.com/user-attachments/assets/0eb21ed9-8e48-4367-a496-698f87057735" /> |
| Workflow - Crawls | <img width="967" alt="Screenshot 2025-05-14 at 4 37 44 PM" src="https://github.com/user-attachments/assets/9b287470-0d9d-45ce-b465-87eac52796e9" /> |
| Workflow List | <img width="342" alt="Screenshot 2025-05-22 at 10 43 08 AM" src="https://github.com/user-attachments/assets/94ca9d94-4381-4c37-aa35-94208f8ec9d6" /> |





<!-- ## Follow-ups -->

SuaYoo committed:
switch to task

SuaYoo committed:
add pause notice

SuaYoo committed:
enable replay

SuaYoo committed:
adjust replay

SuaYoo committed:
update paused dates

SuaYoo committed:
fix type

SuaYoo committed:
show consistent elapsed time

SuaYoo committed:
switch to task

SuaYoo committed:
fix size not showingf

SuaYoo committed:
add name to logs

SuaYoo committed:
paginate crawls

SuaYoo committed:
tweak layout

SuaYoo committed:
simplify poll

SuaYoo committed:
simplify buttons

SuaYoo committed:
show tooltip for entire latest crawl

tw4l commented:
Overall this is working great! As always thank you for the great testing instructions :)

A few small observations:

1. The on-hover tooltip for workflow status in Crawling tab works nicely for when a workflow is paused, but renders a little oddly for other statuses:

<img width="235" alt="Screenshot 2025-05-22 at 11 54 34 AM" src="https://github.com/user-attachments/assets/ac88c416-3569-4010-b80c-22ad5b8770b6" />

2. The buttons for latest crawl (download, copy crawl id, go to archived item) are rather subtle. Even when looking for them because they were listed in the PR description, it took me a little while to locate them the first time. I wonder if grouping them together, making it more visually apparent that they're buttons, or using color might help. I recognize this is similar to what we do for public collections, so it's very possibly fine for this PR. But it might be a good thing to do some user testing on or reconsider at some point in the future? And if anything for this PR, maybe we just have all three buttons on the same line instead of spread out between the sub-tab and title lines?

SuaYoo committed:
update tooltips

SuaYoo committed:
update menu options

SuaYoo committed:
update tooltip

SuaYoo commented:
> The on-hover tooltip for workflow status in Crawling tab works nicely for when a workflow is paused, but renders a little oddly for other statuses

Fixed:

<img width="342" alt="Screenshot 2025-05-22 at 10 43 08 AM" src="https://github.com/user-attachments/assets/d62d07e3-6d9e-4ccc-9cab-e694aa14b631" />

Somewhat out of scope, but also added a tooltip to the modified date for consistency:

<img width="368" alt="Screenshot 2025-05-22 at 10 42 35 AM" src="https://github.com/user-attachments/assets/634e10f0-80e7-451e-a7e1-fb038e341d34" />

> The buttons for latest crawl (download, copy crawl id, go to archived item) are rather subtle.

Updated to show all buttons (when relevant) on the same line, and also added the same actions to the workflow action menu:

<img width="978" alt="Screenshot 2025-05-22 at 12 19 30 PM" src="https://github.com/user-attachments/assets/f0e56d20-3334-4db2-857d-b5d8ad154d1f" />

<img width="320" alt="Screenshot 2025-05-22 at 12 19 45 PM" src="https://github.com/user-attachments/assets/37c540c2-4d1f-47e8-908a-f3317e37e444" />

The "Copy Latest Crawl ID" button is now only in the action menu to lessen its visual impact on the main crawl actions.

ikreymer commented:
Looks good overall! The archived item details icon may still be a bit confusing, as it seems like it'd be downloading something too.. maybe we can add text for 'Details' since there is room there?

SuaYoo committed:
update actions

SuaYoo commented:
Updated to deep link into item sections and standardize tab actions:

<img width="977" alt="Screenshot 2025-05-26 at 3 17 22 PM" src="https://github.com/user-attachments/assets/d2b1920f-5768-4ef8-bc0b-634b126766a8" />

SuaYoo committed:
handle pause

SuaYoo committed:
update labels

SuaYoo committed:
update state renderer

SuaYoo committed:
update state

ikreymer committed:
allow other running states to be set while pausing

SuaYoo committed:
update exec seconds display

SuaYoo committed:
show exec time

SuaYoo committed:
update docs

SuaYoo committed:
fix typo

SuaYoo commented:
Updated to update statuses and show execution time:

<img width="1262" alt="Screenshot 2025-05-27 at 3 28 05 PM" src="https://github.com/user-attachments/assets/05abfc0a-8a9c-4cae-80ce-13e788b42c86" />

emma-sg reviewed (changes requested):
Looks like the updated execution time formatter will render `<0 minutes` when the duration is 0. I believe `detailsRelevant` needs to be taken into account when considering when to include the `"<"` in `compactMinutes`.

emma-sg commented:
~~Oh, could we also remove the three dots to the left of the "More" menu? I don't think it's necessary with the caret to the right of the text.~~

Chatted on Discord, gonna just swap out the three dots I think?

SuaYoo committed:
fix exec time prefix

SuaYoo committed:
replace icon for "More"

ikreymer committed:
return status when finally 'paused'

emma-sg reviewed (approved):
Looks good!

SuaYoo committed:
update states

SuaYoo committed:
hoist tooltip

SuaYoo committed:
use new popover
